### PR TITLE
Check current collision when trajectory could not be generated

### PR DESCRIPTION
**Pull request by 708yamaguchi**
In this pull request, current collision is checked by `/check_state_validity` service when trajectory could not be generated.
If collision is detected, warn message is output.

### CODE REVIEW

File: pr2eus_moveit/euslisp/robot-moveit.l [hunk 1]
```diff
@@ -685,6 +688,21 @@
...
+                 (+ moveit_msgs::PlanningSceneComponents::*ROBOT_STATE*
+                    moveit_msgs::PlanningSceneComponents::*ROBOT_STATE_ATTACHED_OBJECTS*))))
+      (when (not (send res :valid))
+        (warn ";; collision is found in the current state.~%"))
+      ;; contacts are not returned in indigo and earlier versions
+      ;; https://github.com/ros-planning/moveit/pull/2355
+      (when (send res :contacts)
+        (warn ";; contact object pairs:~%")
+        (dolist (c (send res :contacts))
+          (warn ";; ~A~%" (list (send c :contact_body_1) (send c :contact_body_2)))))
```
Comment (knorth55) [inline pr2eus_moveit/euslisp/robot-moveit.l:704]: please use `ros::ros-warn` instead of `warn`.

**Review by knorth55 — CHANGES_REQUESTED** [review]

File: pr2eus_moveit/euslisp/robot-moveit.l (hunk 1 above)
Comment (708yamaguchi) [reply pr2eus_moveit/euslisp/robot-moveit.l]: Thank you, I updated.

Warn logs are like below:
```
3.irteusgl$ send *ri* :check-state-validity
[ WARN] [1605078325.760290920, 25.875000000]: Collision is found in the current state.
Contact object pairs:
(base_link bl_caster_link)
(base_link br_caster_link)
(base_link fl_caster_link)
(base_link fr_caster_link)
#<moveit_msgs::getstatevalidityresponse #X5602ef638be0>
```

**Review by knorth55 — APPROVED** [review]

File: pr2eus_moveit/euslisp/robot-moveit.l (hunk 1 above)
Comment (k-okada) [reply pr2eus_moveit/euslisp/robot-moveit.l]: LGTM, But why caster link collides in this example?

Comment (708yamaguchi) [reply pr2eus_moveit/euslisp/robot-moveit.l]: These collisions only occur on gazebo.
The model of the caster is different between the real fetch and gazebo.

I will create pull request for fetch.srdf to ignore these collisions in gazebo mode.
https://github.com/fetchrobotics/fetch_ros/blob/melodic-devel/fetch_moveit_config/config/fetch.srdf

real fetch urdf:
https://github.com/fetchrobotics/fetch_ros/blob/melodic-devel/fetch_description/robots/fetch.urdf
gazebo fetch urdf:
https://github.com/fetchrobotics/fetch_gazebo/blob/gazebo9/fetch_gazebo/robots/fetch.gazebo.xacro

|real|gazebo|
|---|---|
|||

Comment (708yamaguchi) [reply pr2eus_moveit/euslisp/robot-moveit.l]: I create https://github.com/fetchrobotics/fetch_gazebo/pull/109

**Pull request merged**